### ISSUE TITLE
`setup_rich_logger` interferes with global logging config

### ISSUE THREAD

**Issue by ma-sadeghi** [PMEAL/OpenPNM#2785]
When we set up the customized `rich`/`loguru`, we inadvertantly override the global Python logging format. This should be fixed to only affect the module level (i.e., `opnepnm` or `porespy`). Here's a snippet written by ChatGPT to fix it, I haven't tested it but probably works with little (or no) modification:

```python
import logging

def _setup_logger():
    logger = logging.getLogger('your_library_name')
    logger.setLevel(logging.WARNING)

    formatter = logging.Formatter('- %(levelname)-7s: %(message)s \n'
                                  '- SOURCE : %(name)s.%(funcName)s \n'
                                  '- TIME   : %(asctime)s')

    handler = logging.StreamHandler()
    handler.setFormatter(formatter)
    logger.addHandler(handler)

    del formatter, handler

def _setup_logger_rich():
    logger = logging.getLogger('your_library_name')
    logger.setLevel(logging.WARNING)

    from rich.logging import RichHandler

    handler = RichHandler(rich_tracebacks=True)
    logger.addHandler(handler)
```

**Comment by ma-sadeghi** [comment]
See https://github.com/PMEAL/porespy/issues/891

**Comment by ma-sadeghi** [comment]
Fixed in #2531
